File: Python/0351-0400/389. Find the Difference.py
#  https://leetcode.com/problems/find-the-difference/

class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        result = 0

        for letter in s: result ^= ord(letter) #ord is ASCII value
        for letter in t: result ^= ord(letter)

        return chr(result) #convert ASCII into character

File: Javascript/0351-0400/389. Find the Difference.js
// https://leetcode.com/problems/find-the-difference/

var findTheDifference = function(s, t) {
    let result = 0;

    for (let i = 0; i < s.length; i++) 
        result ^= s.charCodeAt(i);

    for (let i = 0; i < t.length; i++) 
        result ^= t.charCodeAt(i);
    
    return String.fromCharCode(result);
};
